src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzer.Value.cs

```diff
@@ -0,0 +1,244 @@
+                        var platformName = SwitchVersionedPlatformName(invokedPlatformCheckMethod.Name);
+
+                        if (platformName != null && TryDecodeOSVersion(arguments, valueContentAnalysisResult, out var version))
+                        {
+                            info = new RuntimeMethodValue(invokedPlatformCheckMethod.Name, platformName, version, negated: false);
+                            return true;
+                        }
+                    }
+                }
+
+                info = default;
+                return false;
+            }
+
+            private static string? SwitchVersionedPlatformName(string methodName)
+            => methodName switch
+            {
+                IsWindowsVersionAtLeast => Windows,
+                IsMacOSVersionAtLeast => MacOS,
+                IsIOSVersionAtLeast => IOS,
+                IsAndroidVersionAtLeast => Android,
+                IsFreeBSDVersionAtLeast => FreeBSD,
+                IsTvOSVersionAtLeast => TvOS,
+                IsWatchOSVersionAtLeast => WatchOS,
+                _ => null
+            };
+
+            private static string? SwitchPlatformName(string methodName)
+                 => methodName switch
+                 {
+                     IsWindows => Windows,
+                     IsLinux => Linux,
+                     IsMacOS => MacOS,
+                     IsIOS => IOS,
+                     IsBrowser => Browser,
+                     IsAndroid => Android,
+                     IsFreeBSD => FreeBSD,
+                     IsTvOS => TvOS,
+                     IsWatchOS => WatchOS,
+                     _ => null
+                 };
+
+            private static bool TryDecodeRuntimeInformationIsOSPlatform(
+                IOperation argumentValue,
+                INamedTypeSymbol osPlatformType,
+                [NotNullWhen(returnValue: true)] out string? osPlatformName)
+            {
+                if ((argumentValue is IPropertyReferenceOperation propertyReference) &&
+                    propertyReference.Property.ContainingType.Equals(osPlatformType))
+                {
+                    osPlatformName = propertyReference.Property.Name;
+                    return true;
+                }
+
+                osPlatformName = null;
+                return false;
+            }
+
+            private static bool TryDecodeOSVersion(
+                ImmutableArray<IArgumentOperation> arguments,
+                ValueContentAnalysisResult? valueContentAnalysisResult,
+                [NotNullWhen(returnValue: true)] out Version? osVersion,
+                int skip = 0)
+            {
+
+                using var versionBuilder = ArrayBuilder<int>.GetInstance(4, fillWithValue: 0);
+                var index = 0;
+
+                foreach (var argument in arguments.Skip(skip))
```

Comment:
I just realized this will not work as expected for named arguments in random order. Arguments are in order of evaluation, not the parameter order. You should write an extension method on Arguments that transforms them from evaluation order to parameter order. Please also add some basic unit tests for all different attributes and guard method calls where some named arguments are used with changed order. Additionally, given some methods/attribute constructors  take optional arguments we should add tests where subset of optional arguments are provided, some intermediate optional argument is skipped, etc.

Reply:
That is good point

Reply:
I added this method for finding a single argument operation at a specific parameter index: https://github.com/dotnet/roslyn-analyzers/blob/8bc7b0bf842d3fee3ee8595e783828f8c4ee448e/src/Utilities/Compiler/Extensions/IOperationExtensions.cs#L768-L785 

You can probably write up some similar logic that returns re-ordered arguments array